### PR TITLE
feat: MongoDBAtlas Document Store

### PR DESCRIPTION
**Pull request by ZanSara**
Related to #143 

Add support for MongoDB Atlas in the form of `MongoDBAtlasDocumentStore`.

Note that:
- Filters are not implemented yet
- No retrievers are implemented yet

The missing features will be added in later PRs.

### CODE REVIEW

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:87]: Nitpick - Let's extract the argument into its own variable. Will make it easier to debug.

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:103]: Why do we need to do this? Let's add a comment with the explanation.

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:64]: Why don't we cache the output in the component and use it instead? The index lookup described in the docstring is not implemented here, after all.

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:54]: This only seems to be used in the ctor, so we don't have to store it?

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py:8]: Wouldn't it make sense to raise an error here instead? Silently failing (even with a warning) when the user was expecting otherwise is something we should avoid.

File: integrations/mongodb_atlas/tests/test_document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/tests/test_document_store.py:55]: Missing `from_dict` test.

**Review by shadeMe — CHANGES_REQUESTED** [review]

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (ZanSara) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:64]: Good point! That's more of a leftover than anything, I'll remove it

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py
Comment (ZanSara) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py:8]: Right, good point!

File: integrations/mongodb_atlas/tests/test_document_store.py
Comment (ZanSara) [reply integrations/mongodb_atlas/tests/test_document_store.py:55]: from_dict is not reimplemented in MongoDBAtlasDocumentStore, so I think it's safe not to test

File: integrations/mongodb_atlas/pyproject.toml
Comment (shadeMe) [inline integrations/mongodb_atlas/pyproject.toml:28]:
```suggestion
  "haystack-ai>=2.0.0b6",
```

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:53]:
```suggestion
```
This is dead code as `resolve_value` will raise an error if the env var is not set (by default).

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py
Comment (shadeMe) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py:2]: Type annotation.

**Review by shadeMe — CHANGES_REQUESTED** [review]
Good to merge after incorporating the following changes 👍

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py
Comment (ZanSara) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/filters.py:2]:
```suggestion
def haystack_filters_to_mongo(filters: Optional[Dict[str, Any]]):
```

**Review by shadeMe — APPROVED** [review]

File: integrations/mongodb_atlas/pyproject.toml
Comment (masci) [inline integrations/mongodb_atlas/pyproject.toml:42]: This tag is wrong

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (masci) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:89]: This doesn't respect the protocol: https://github.com/deepset-ai/haystack/blob/main/haystack/document_stores/types/protocol.py#L38

File: integrations/mongodb_atlas/pyproject.toml
Comment (ZanSara) [reply integrations/mongodb_atlas/pyproject.toml:42]: This was fixed right after the merge: https://github.com/deepset-ai/haystack-core-integrations/pull/426

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py
Comment (ZanSara) [reply integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py:89]: https://github.com/deepset-ai/haystack-core-integrations/pull/430

**Pull request merged**
